include/swift/AST/DiagnosticsParse.def

```diff
@@ -724,6 +724,9 @@ ERROR(rethrows_in_wrong_position,none,
       "'rethrows' may only occur before '->'", ())
 ERROR(throw_in_function_type,none,
       "expected throwing specifier; did you mean 'throws'?", ())
+ERROR(throw_in_function_type_and_wrong_position,none,
+      "expected throwing specifier; did you mean 'throws'? "
+      "'throws' may only occur before '->'", ())
```

Comment:
This should be in a separate note which applies a fix-it. You could then reuse the `throw_in_function_type` diagnostic. Also, it might be better to not have a separate not for this at all, but just reuse `rethrows_in_wrong_position` diagnostic. You can tweak it so it's `%select{'throws'|'rethrows'}0 may only occur before '->'` and then pass `1` when emitting this diagnostic.